aws-codeguruprofiler-profilinggroup/src/main/java/software/amazon/codeguruprofiler/profilinggroup/DeleteHandler.java

```diff
@@ -28,15 +30,18 @@
 
         final ResourceModel model = request.getDesiredResourceState();
         final String awsAccountId = request.getAwsAccountId();
+        final String profilingGroupName = model.getProfilingGroupName();
 
         try {
             DeleteProfilingGroupRequest deleteProfilingGroupRequest = DeleteProfilingGroupRequest.builder()
-                    .profilingGroupName(model.getProfilingGroupName())
+                    .profilingGroupName(profilingGroupName)
                     .build();
 
             proxy.injectCredentialsAndInvokeV2(deleteProfilingGroupRequest, profilerClient::deleteProfilingGroup);
 
-            logger.log(String.format("%s [%s] for accountId [%s] has been successfully deleted!", ResourceModel.TYPE_NAME, model.getProfilingGroupName(), awsAccountId));
+            deleteNotificationChannelsForProfilingGroup(profilingGroupName, proxy, profilerClient);
+
+            logger.log(String.format("%s [%s] for accountId [%s] has been successfully deleted!", ResourceModel.TYPE_NAME, profilingGroupName, awsAccountId));
```

Comment:
Both me and Lambros already asked about this -- my understanding was that we did not need to delete the notification configuration before we can delete a profiling group. Can you look into it and clarify? Because otherwise we [also have a documentation gap](https://docs.aws.amazon.com/codeguru/latest/profiler-api/API_DeleteProfilingGroup.html) on that operation, since that's not mentioned there as well.

Reply:
So yes, you do not need to remove the notification configuration in order to delete a profiling group, so I will just remove this then.